### PR TITLE
[CategoryAxis] - Correcting textFits parameter for space calculations

### LINKED ISSUE #1511: XAxes overlapping (repro/analysis in progress)
NO repro yet, attempt: http://jsfiddle.net/xL0duLo9/

EDIT: Changing the svg height on the above example also messes the chart up
http://jsfiddle.net/xL0duLo9/6/

Looks like this in Slate: 
screen shot 2015-01-09 at 5 29 20 pm

Only happens for ordinal scales.

EDIT (@bluong) : Repro has been made.  Simplifying the example to ensure it's mainly an axis issue, you see the corresponding image:

screen shot 2015-01-22 at 12 25 28 pm

Corresponding jsfiddle - http://jsfiddle.net/xL0duLo9/4/

NOTE: If you remove any of the Label components, the axes get messed up even more, but if you remove both, it acts fine.

EDIT (@bluong) : Even nicer, simpler repro
http://jsfiddle.net/xL0duLo9/8/

### PR DESCRIPTION
**Pull request by bluong**
The textFits parameter currently determines if the text labels on a categoryAxis can fit there.  However the calculation is... incorrect.

It used to read - If I have 1 wrapped line and I am not not an empty string, then I fit
It should read - If I have 1 wrapped line and I am not an empty string, then I fit

In short, confusing double negative is confusing

Fixes #1511

### CODE REVIEW

**Comment by jtlan** [comment]
Test?

File: test/components/categoryAxisTests.ts [hunk 1]
```diff
@@ -108,4 +108,22 @@ describe("Category Axes", () => {
 
     svg.remove();
   });
+
+  it("layout calculated correctly under 0 width/height surrounding component conditions", () => {
```
Comment (jtlan) [inline test/components/categoryAxisTests.ts]: Is there a test that doesn't involve `Table`? What was the root cause of the bug?

**Comment by jtlan** [comment]
Can we get a more "unit" test?

**Comment by jtlan** [comment]
Do we also want a corresponding test for a vertical `Axis.Category`?

File: test/components/categoryAxisTests.ts [hunk 2]
```diff
@@ -108,4 +108,19 @@ describe("Category Axes", () => {
...
+    var svg = generateSVG(300, 300);
+    var years = ["2000", "2001", "2002", "2003"];
+    var scale = new Plottable.Scale.Ordinal().domain(years);
+    var axis = new Plottable.Axis.Category(scale, "bottom");
+    axis.renderTo(svg);
+    var requestedSpace = axis._requestedSpace(300, 10);
+    assert.isTrue(requestedSpace.wantsHeight, "axis should ask for more space");
+    axis.orient("left");
+    requestedSpace = axis._requestedSpace(10, 300);
+    assert.isTrue(requestedSpace.wantsWidth, "axis should ask for more space");
```
Comment (jtlan) [inline test/components/categoryAxisTests.ts]: Can we distinguish between the two asserts in the message? "axis should ask for more space (vertical/horizontal)"

**Comment by jtlan** [comment]
[minor] Distinguishing `assert`s.

**Comment by crmorford** [comment]
In some cases, the axis seems to be grabbing a lot of extra vertical space. Is that intentional?
The "category_axis_non_string" quicktest is a good example

**Comment by crmorford** [comment]
Verified it fixes slate bug though. Just want to see if you know why it's taking extra space sometimes.

**Comment by bluong** [comment]
So essentially unless I'm mistaken, it's taking up extra space because of the gutter concept, which is defaulted to 15 pixels I believe.

**Comment by crmorford** [comment]
GIL

**Pull request merged**